File: python/code_challenges/tree_fizz_buzz/tree_fizz_buzz.py

class Node:
    def __init__(self, value = None, left = None, right = None):
        self.value = value
        self.left = left
        self.right = right

class K_aryTree:

    def __init__(self, root = None):
        self.root = root

    def pre_order(self):
        list = []

        def traverse(root):

            if root is None:
                return "Empty Tree"

            list.append(root.value)
            traverse(root.left)
            traverse(root.right)

        traverse(self.root)
        return list


# @staticmethod
def fizz_buzz_tree(tree):

    root = tree.root
    list = []

    if root.value is None:
      return "Empty Tree"

    def traverse(root):

        if root is None:
           return
        if root.value % 3 == 0 and root.value % 5 == 0:
            root.value = "FizzBuzz"
        elif root.value % 3 == 0:
            root.value == "Fizz"
        elif root.value % 5 == 0:
            root.value = "Buzz"
            list.append(root.value)
            traverse(root.left)
            traverse(root.right)

        traverse(root)

File: javascript/code-challenges/trees/src/code-challenges/tree-fizz-buzz/tree-fizz-buzz.js
// const { root } = require('cheerio/lib/static.js');
const BinaryTree = require('../../binaryTree.js');
const KNode = require('../../kNode.js');
// const KNode = require('../../kNode.js');

function fizzBuzzTree(kt){

  if(kt.root === undefined){
    throw new Error('input tree is empty');
  }
  let newTree = new BinaryTree();
  newTree.root = kt.root;


  function traverse(root){
    fizzBuzz(root);
    if(root.children){
      root.children.forEach(childNode => {
        traverse(childNode);
      });
    }
  }

  function fizzBuzz(node){
    if(node.value % 3 === 0 && node.value % 5 === 0){
      node.value = 'FizzBuzz';
    } else if(node.value % 5 === 0){
      node.value = 'Buzz';
    } else if(node.value % 3 === 0){
      node.value = 'Fizz';
    } else {
      node.value = node.value.toString();
    }
  }

  function print(tree){
    let elements = [];
    if(tree.root === undefined){
      throw new Error('binary tree is empty');
    }
    function printTraverse(root){
      elements.push(root.value);
      if(root.children){
        root.children.forEach(child => {
          elements.push(child.value);
          traverse(child);
        });
      }
    }
    printTraverse(tree.root);
    return elements;
  }

  traverse(newTree.root);
  let nodes = print(newTree);
  return nodes;
}

module.exports = fizzBuzzTree;
